src/resolve.rs

```diff
@@ -14,22 +14,23 @@ use std::{
 };
 
 impl Resolver {
-    pub(crate) fn append_ext_for_path(path: &Path, ext: &str) -> PathBuf {
-        PathBuf::from(&format!("{}{ext}", path.display()))
-    }
-
     fn resolve_file_with_ext(&self, path: PathBuf, info: Info) -> State {
+        let mut path = path.display().to_string();
         for ext in &self.options.extensions {
-            let path = Self::append_ext_for_path(&path, ext);
-            if self.load_entry(&path).is_file() {
+            let len = ext.len();
+            path.push_str(ext);
```

Comment:
If you know the number of allocations beforehand, you should reserve the exact amount.